packages/hardhat-core/src/internal/hardhat-network/provider/modules/eth.ts

```diff
@@ -590,9 +592,18 @@ export class EthModule {
       filter.toBlock = new BN(block.header.number);
     }
 
+    const [fromBlock, toBlock] = await Promise.all([
```

Comment:
I don't understand this change. It isn't related to the rest of the changes, is it? (That's ok by me if it's an improvement; just trying to understand it).

Reply:
It's related. This used to be synchronous so it was fine to call the function serially. However, now that it's asynchronous I refactored it so it's called asynchronously in parallel for performance reasons.

Reply:
Oh, I see :+1: